### PR TITLE
Configure non secrets environment variable into yml file

### PR DESCRIPTION
**Pull request by felixtheflex**
### Context

In preparation for multi-stage (environment) deployment, the app specific environment variables are stored in yml file

### Changes proposed in this pull request

Environment specific variables for each stage e.g. QA, Staging or Production are now emcompassed into a yml list

### CODE REVIEW

File: terraform/terraform.tf [hunk 1]
```diff
@@ -39,4 +39,5 @@ module paas {
   log_url                     = var.paas_log_url
   docker_credentials          = local.docker_credentials
   app_secrets_variable        = local.app_secrets
+  app_config_variable         = local.app_config
```
Comment (vigneshmsft) [inline terraform/terraform.tf:42]: you could `merge` the two and pass as app_env_variables to the paas module

File: terraform/modules/paas/variables.tf [hunk 2]
```diff
@@ -25,21 +25,17 @@ variable docker_credentials { type = map }
...
-    RAILS_ENV                      = "production"
-    RACK_ENV                       = "production"
-    RAILS_SERVE_STATIC_FILES       = true
-    SECRET_KEY_BASE                = "TestKey"
-  }
+  app_environment_config = var.app_config_variable
   app_environment_secrets = var.app_secrets_variable
 
-  app_environment = merge(local.app_environment_none_secrets, local.app_environment_secrets)
+  app_environment = merge(local.app_environment_config, local.app_environment_secrets)
```
Comment (vigneshmsft) [inline terraform/modules/paas/variables.tf:38]:
```suggestion
  app_environment = merge(var.app_environment_config, local.app_environment_secrets)
```
if not merging in the root module, you could remove `local.app_environment_config` and just use `var`

File: terraform/terraform.tf (hunk 1 above)
Comment (felixtheflex) [reply terraform/terraform.tf:42]: the merge is done in the module itself...

File: terraform/modules/paas/variables.tf (hunk 2 above)
Comment (felixtheflex) [reply terraform/modules/paas/variables.tf]: yes that could work too...

**Review by defong — APPROVED** [review]

**Review by vigneshmsft — APPROVED** [review]

**Review by misaka — APPROVED** [review]
LGTM

**Pull request merged**